runtime/sema/type.go

```diff
@@ -3369,6 +3423,18 @@ func (t *FunctionType) initializeMemberResolvers() {
 	})
 }
 
+func (t *FunctionType) CheckInstantiated(pos ast.HasPosition, memoryGauge common.MemoryGauge, report func(err error)) {
```

Comment:
Could you please add a test case for a function type with type parameters?

Such a function cannot be defined in a Cadence program, so you'll need to define it in Go and inject it

Reply:
Can you please guide me on how to write such a test? This check is done on function definition. So if we cannot define it in Cadence program, then not sure how to add the test case.

I tried the following:

```
	t.Run("Struct with function with type parameters", func(t *testing.T) {

		t.Parallel()

		_, err := parseAndCheckWithTestValue(t,
			`
				struct Bar {
					init() {
						let f = test
					}
				}
			`,
			stdlib.InclusiveRangeConstructorFunctionType,
		)
		require.NoError(t, err)
	})
```

The function being injected is the `InclusiveRange<T>` constructor function.

Reply:
Added tests in 89c1a6008dc212f18a1a178c30d88c9f51294aee